### ISSUE TITLE
JEI not showing recipes from other mods on a online server

### ISSUE THREAD

**Issue by Insomniakh** [mezz/JustEnoughItems#2135]
Hello all,
I have a base server to play with a friend. We added a few mods to the server (JEI included of course, alond some furniture/backpack mods). 

Here's the problem: when we play our server, key U (Uses) and key R (Recipes) when hovering over an item doesn't work with MODDED items (like Storage Drawners or Useful Backpacks), but it works well with Vanilla items. I can craft them manually, so all the mods works fine, but I need to search for the recipe on the mod's website or something... In some items when I press R, I can see the "information" tab, but not how to create them. We played in another server with exactly ALL the mods and the same forge version, and it worked well, but not in my case. When we play offline, it works well too. 

I already reinstalled the server/minecraft/java/updated/downgraded/and so on, and nothing works. I don't know why it happens, and I can't find the answer on internet...
Can u help me please? Thanks in advance <3

PS: Sorry if I have some grammar errors, my main language isn't English, but I did my best ;)

**Comment by Ommina** [comment]
A quick clarification: when you say "when we play offline, it works well too", do you mean the server is in offline mode (which is an authentication thing, and seems unlikely) or do you mean it works in a local, single-player world?

If you go into a single-player world and it works, THEN logon to the server, does it continue to work?

If so, you may find #2101 helpful.

**Comment by Insomniakh** [comment]
Hi Ommina! Thanks for the comment
I was trying to say that when I play singleplayer, the mod works as intended, showing me all the other mods recipes; but as soon as I connect to my server, then JEI isn't showing me that recipes, just the vanilla one. So the mod works "fine" on single player but not on MP.
I just had read #2101, but the problem isn't exactly that. Anyway I'll try all the possible solutions!
I'm attaching 2 screenshots; for the first one I have created a singleplayer world to show you how the recipe of the backpack is showed fine (in spanish; I have already tried to change the language to english but it doesn't affect the result); after that, I joined my own server (to clarify: my server is the normal server that u can download on minecraft website freely) and I couldn't see the same recipe.
Hope it helps! 
![2020-10-25_21 35 34](https://user-images.githubusercontent.com/73425743/97118675-a1b4ec80-170b-11eb-9c1d-2bc7b46a44a4.png)
![2020-10-25_21 45 55](https://user-images.githubusercontent.com/73425743/97118679-a37eb000-170b-11eb-9c86-d3c6796bb289.png)

**Comment by paladest** [comment]
Have the same problem, did you solve it?

**Comment by Insomniakh** [comment]
> Have the same problem, did you solve it?

Nop, I didn't... I discovered that it should be a problem not with JEI itself, but with the minecraft base server from the official page. 
The server does accept all the mods, but doesn't create a config file for anyone of them, so basically you "have" the mods in the server, but doesn't work well because there's no configuration folder/files anywhere. The server just loads with the base files and nothing more than that.
I ended up creating an online server with a mediocre but free hosting server, at least all the mods works fine.
I'm closing this issue because it's not a JEI issue itself, just a problem when using it on the base minecraft server. Thanks anyway!
